playwright/src/main/java/com/microsoft/playwright/impl/BrowserTypeImpl.java

```diff
@@ -60,7 +61,8 @@ public Browser connect(String wsEndpoint, ConnectOptions options) {
   private Browser connectImpl(String wsEndpoint, ConnectOptions options) {
     try {
       Duration timeout = Duration.ofDays(1);
-      Map<String, String> headers = Collections.emptyMap();
+      Map<String, String> headers = new HashMap<>();
+      headers.put("User-Agent", getUserAgent());
```

Comment:
I see two problems with this:
1. This way the user-agent field is going to be always overridden (at line are 72) if any headers are passed in the options. We want it to be overridden only if 'User-Agent' header is present in the options.
2. The header name should be matched case-insensitively, i.e. 'user-agent'  header option should still override default value.

Reply:
Updated